src/ui/component/toolbar/toolbar.tsx

```diff
@@ -26,32 +115,69 @@ const BurgerMenu = ({ expandSidebar, theme }: BurgerProps): JSX.Element => (
   </div>
 )
 
-const WalletButton = ({ isSmallScreen, label }: WalletButtonProps): JSX.Element => (
-  <div className="okp4-dataverse-portal-wallet">
-    {isSmallScreen ? (
-      <Button
-        iconButtonOnly={
-          <div className="okp4-dataverse-portal-wallet-icon">
-            <Icon name="wallet" />
-          </div>
-        }
-        variant="tertiary"
-      />
-    ) : (
-      <Button
-        icons={{ startIcon: <Icon name="wallet" /> }}
-        label={label}
-        size="large"
-        variant="tertiary"
-      />
-    )}
-  </div>
-)
+const WalletButton = ({
+  isWalletConnected,
+  isSmallScreen,
+  isWalletMenuOpen,
+  label,
+  onWalletMenuClick
+}: WalletButtonProps): JSX.Element => {
+  const theme = useAppStore(state => state.theme)
 
+  const walletIcons = useMemo(
+    () => ({
+      startIcon: <Icon name="wallet" />,
+      endIcon: isWalletConnected ? (
+        <div
+          className={classNames(
+            `okp4-dataverse-portal-wallet-expand ${isWalletMenuOpen ? 'rotate-down' : 'rotate-up'}`
+          )}
+        >
+          <Icon name={`expand-down-${theme}`} />
+        </div>
+      ) : undefined
+    }),
+    [isWalletConnected, isWalletMenuOpen, theme]
+  )
+
+  return (
+    <div className="okp4-dataverse-portal-wallet">
+      {isSmallScreen ? (
+        <Button
+          iconButtonOnly={
+            <div className="okp4-dataverse-portal-wallet-icon">
+              <Icon name="wallet" />
+            </div>
+          }
+          variant="tertiary"
+        />
+      ) : (
+        <Button
+          icons={walletIcons}
+          label={label}
+          onClick={onWalletMenuClick}
+          size="large"
+          variant="tertiary"
+        />
+      )}
+    </div>
```

Comment:
@0xLodz  It is a good point but I'm not sure user agent sniffing is more reliable for our need. [Verifying features could be a more appropriate way](https://developer.mozilla.org/en-US/docs/Web/HTTP/Browser_detection_using_the_user_agent).

Reply:
@ErikssonJoakim I think there is no silver bullet. We want to determine which type of device the browser runs on. Which is what they are talking about in [this section](https://developer.mozilla.org/en-US/docs/Web/HTTP/Browser_detection_using_the_user_agent#mobile_tablet_or_desktop) from the MDN link you shared. I think feature detection are not more reliable for our specific need, e. g. Miscrosoft Surface handle multiple touch points.

Reply:
Currently, the mobile view concerning the connection to the wallet is not yet implemented, it will be part of another ticket. 
This part of code has been moved but was already present.
So I think the comments are not relevant anymore (for this PR), but can be considered for the next ticket.